plugins/modules/azure_rm_adapplication.py

```diff
@@ -146,33 +146,103 @@
     optional_claims:
         description:
             - Declare the optional claims for the application.
-        type: list
-        elements: dict
+        type: dict
         suboptions:
-            name:
-                description:
-                    - The name of the optional claim.
-                type: str
-                required: True
-            source:
+            access_token:
```

Comment:
keep same with argument spec
```suggestion
            access_token_claims :
```